mlflow/models/evaluation/default_evaluator.py

```diff
@@ -79,7 +80,9 @@ def _infer_model_type_by_labels(labels):
 def _extract_raw_model(model):
     model_loader_module = model.metadata.flavors["python_function"]["loader_module"]
     if model_loader_module == "mlflow.sklearn" and not isinstance(model, _ServedPyFuncModel):
-        return model_loader_module, model._model_impl.sklearn_model
+        if isinstance(model._model_impl, _SklearnModelWrapper):
+            return model_loader_module, model._model_impl.sklearn_model
```

Comment:
Can we add a comment on why we need this?

Reply:
I'm also curious about what happens if we evaluate a non-sklearn model like xgboost.

Reply:
I think there're two problems:
1. https://github.com/mlflow/mlflow/blob/4f76275317accb0b51770a4a83b8d1db2fda21f9/mlflow/models/evaluation/default_evaluator.py#L93-L103 The raw_model will never be xgboost.XGBModel as it will be the wrapper, and predict_prob_fn will be None for all wrappers as they're not implemented.
2. https://github.com/mlflow/mlflow/blob/4f76275317accb0b51770a4a83b8d1db2fda21f9/mlflow/models/evaluation/default_evaluator.py#L730-L732 Looks like `shap.Explainer` doesn't accept wrappers